express-api/src/services/buildings/buildingSchema.ts

```diff
@@ -3,7 +3,9 @@ import z from 'zod';
 export const BuildingFilterSchema = z.object({
   pid: z.coerce.number().nonnegative().optional(),
   classificationId: z.coerce.number().nonnegative().optional(),
-  agencyId: z.coerce.number().nonnegative().optional(),
+  agencyId:
+    z.coerce.number().nonnegative().optional() ||
+    z.array(z.number().int().nonnegative()).optional(),
```

Comment:
I had to check to make sure this works because it's not a documented way to do unions, but it does work... Pretty neat. I think that it may only work due to the first schema being `optional()`, which I think means that side of the logical or will evaluate to undefined when it fails? So may behave differently with non-optional schemas, just something to keep in mind.